### ISSUE TITLE
Geocoder::Lookup::Test stubbing failing after upgrading from 1.1.8 to 1.2.9

### ISSUE THREAD

**Issue by pzaich** [alexreisner/geocoder#885]
I am in the process of upgrading Geocoder to the latest version to take advantage of the improvements made handle caching related to Bing referenced by #840. However, after upgrading, my Test configuration no longer appears to be working. Real geocoding requests are being made instead. Rails 3.2.21.

I see logging messages like this: 
```
Geocoder: HTTP request being made for http://dev.virtualearth.net/REST/v1/Locations?q=Funkchester,%20RI%20(46339)&key=xyz
```
https://github.com/alexreisner/geocoder/blob/master/lib/geocoder/lookups/base.rb#L186

```
# rails_helper.rb
Geocoder.configure(lookup: :test)
Geocoder::Lookup::Test.add_stub(
  "not real", []
)
Geocoder::Lookup::Test.set_default_stub(
  [
    {
      'latitude'     => 37.724282,
      'longitude'    => -122.456834,
      'address'     => 'San Francisco, CA, USA',
      'state'        => 'San Francisco',
      'state_code'   => 'CA',
      'country'      => 'United States',
      'country_code' => 'US',
      'Postal'       => '94110'
    }
  ]
)
```
Adding this configuration directly into the console falls back to this default stub correctly. 

```
#geocoder.rb

Geocoder.configure(
  timeout: 10,
  lookup: :bing,
  api_key: ENV['BING_MAPS_API_KEY'],
  ip_lookup: :maxmind,
  cache: Rails.cache,
  maxmind: {
    api_key: ENV['MAXMIND_KEY'],
    service: :city_isp_org
  }
)
```

**Comment by alexreisner** [comment]
I can't seem to duplicate this. Can you give me more information about your test environment configuration? Also, what's the significance of [line 186](https://github.com/alexreisner/geocoder/blob/master/lib/geocoder/lookups/base.rb#L186)?

**Comment by pzaich** [comment]
Alex,
I've been able to identify this as an error in our codebase. Interestingly the error didn't crop up until we made the upgrade to 1.2.9.

We were using a custom lookup in our model that hard-coded to :bing unless a custom lookup was defined on an attribute on the model

```
class Address < ActiveRecord::Base
  geocoded_by :full_street_address, lookup: proc { |obj| obj.geocoder_lookup }

  def geocoder_lookup
    custom_lookup.try(:to_sym) || :bing
  end
end
```

It looks like previously, Geocoder stubbed over this method, but it no longer does. (not sure why)The fix for us was to change this method to default to the Gecoder config lookup.
```
  def geocoder_lookup
    custom_lookup.try(:to_sym) || Geocoder.config[:lookup]
  end
```
Then this lookup falls back to :test on our test environment.
